src/interpolator_avx.rs

```diff
@@ -10,85 +11,44 @@ use core::arch::x86_64::{
 use core::arch::x86_64::{
     _mm256_fmadd_ps, _mm256_loadu_ps, _mm256_setzero_ps, _mm_add_ps, _mm_hadd_ps,
 };
-use std::marker::PhantomData;
-
 use crate::asynchro::SincInterpolator;
-
-/// An AVX accelerated interpolator
-pub struct AvxInterpolator<T> {
-    sincs_s: Option<Vec<Vec<__m256>>>,
-    sincs_d: Option<Vec<Vec<__m256d>>>,
-    length: usize,
-    nbr_sincs: usize,
-    phantom: PhantomData<T>,
-}
-
-impl SincInterpolator<f32> for AvxInterpolator<f32> {
-    /// Calculate the scalar produt of an input wave and the selected sinc filter
-    fn get_sinc_interpolated(&self, wave: &[f32], index: usize, subindex: usize) -> f32 {
-        assert!((index + self.length) < wave.len());
-        assert!(subindex < self.nbr_sincs);
-        unsafe { self.get_sinc_interpolated_unsafe(wave, index, subindex) }
-    }
-
-    fn len(&self) -> usize {
-        self.length
-    }
-
-    fn nbr_sincs(&self) -> usize {
-        self.nbr_sincs
-    }
-}
-
-impl SincInterpolator<f64> for AvxInterpolator<f64> {
-    /// Calculate the scalar produt of an input wave and the selected sinc filter
-    fn get_sinc_interpolated(&self, wave: &[f64], index: usize, subindex: usize) -> f64 {
-        assert!((index + self.length) < wave.len());
-        assert!(subindex < self.nbr_sincs);
-        unsafe { self.get_sinc_interpolated_unsafe(wave, index, subindex) }
-    }
-
-    fn len(&self) -> usize {
-        self.length
-    }
-
-    fn nbr_sincs(&self) -> usize {
-        self.nbr_sincs
-    }
+use crate::Sample;
+
+/// Collection of cpu features required for this interpolator.
+static FEATURES: &[CpuFeature] = &[CpuFeature::Avx, CpuFeature::Fma];
```

Comment:
This could be simplified. There has ever been a cpu with fma, that doesn't also have avx. And since we require fma, it's enough to report if that is missing. If avx isn't supported, fma isn't either. Then the list of features isn't needed. 

Anyway right now it's returning that both are missing even if avx is available, on for example a sandy bridge that has avx but not fma.

Reply:
Yeah, it was just intended to do more-or-less the same thing in case you unwrap:ed the error as the panic previously did. We can definitely make it smarter and inform you of the exact sequence of features missing. All though then the error would have to be dynamic.

I'm not sure about feature flags tbh. Recent rustc experience has shaken my belief in how stable the assumptions are from cpu to cpu. But I'll defer to what you want to do here.

What would you want to change?

Reply:
The idea was that if we only care about fma, then the cpu features list will always contain only one element. And then it doesn't need to be a list. That in turn means that CpuFeature and MissingCpuFeatures could be merged to just MissingCpuFeature.
Of course when checking for fma, it should still also check for avx to make sure.

Reply:
All righty, done!